File: src/main/java/com/rang/scheduler/entities/Patient.java
package com.rang.scheduler.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.rang.scheduler.entities.utils.Status;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;

@Entity(name="Patient")
@Table(name="patient")
@Getter
@Setter
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer"})
public class Patient implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String lastName;
    private LocalDate register;//need local date for age or int age
    private String phone;
    private String email;
    private int age;

    private String dni;

    @ManyToOne
    @JoinColumn(name = "address_id")
    private Address address;

    @ManyToOne
    @JoinColumn(name = "agent_id")
    private Agent agent;

    @Enumerated(EnumType.STRING)
    private Status status;





}

File: scheduler/src/app/models/Patient.ts
export interface Patient{
    id:number;
    name:string;
    lastName:string;
    register:string|Date;
    address:Address;
    agent?: Agent;
    age:number;
    status:Status;

    phone:number;
    email:string;
}

export interface Agent {
    name?: string;
    image?: string;
}

export interface Address {
    state?: string;
    description?: string;
}

export enum Status{
    HIGHER,MEDIUM,LOW,OTHER
    
}